pkg/api/validate/dynamic/dynamic.go

```diff
@@ -389,6 +389,11 @@ func (dv *dynamic) ValidateSubnets(ctx context.Context, oc *api.OpenShiftCluster
 			}
 		}
 
+		if ss.SubnetPropertiesFormat == nil ||	
+			    if ss.SubnetPropertiesFormat.ProvisioningState != mgmtnetwork.Succeeded {
```

Comment:
This won't compile with the second `if` inlined here. Once this is resolved it looks like there will be some tests that need updating in `pkg/api/validate/dynamic_test.go` specifically the test function `TestValidateSubnets` on line 558 with the tests named:
* "pass", 
* "pass (master subnet)", 
* "pass (cluster in creating provisioning status)", 
* "fail: invalid subnet CIDR", "fail: too small subnet CIDR"


While updating these tests to pass, please consider adding a new test for the new condition/error message which can be returned.
```suggestion
			     ss.SubnetPropertiesFormat.ProvisioningState != mgmtnetwork.Succeeded {
```

Reply:
Yup rechecked and removed unnecessary if, and added tests too.
Please let me know in case of any changes are needed. Thank you!